enhancements/machine-api/control-plane-machine-set.md

```diff
@@ -0,0 +1,1090 @@
+
+This in turn means, that in a UPI (where typically Machine objects do not exist) or misconfigured clusters, a user
+adding a `ControlPlaneMachineSet` will result in a degraded cluster.
+Users will need to remove the invalid `ControlPlaneMachineSet` resource, or manually add correctly configured Control
+Plane Machines to their cluster to restore their cluster to a healthy state.
+
+We do not recommend that UPI users attempt to adopt their control plane instances into Machine API due to the
+likelihood that the resulting spec does not match the existing Machines. Instead, we recommend that UPI users initially
+go through the process of replacing their control plane instances with Machines by creating new Control Plane Machines,
+allowing Machine API to create the instances, and then removing their old, manually created control plane instance.
+This effectively means migrating their entire control plane onto Machine API before the `ControlPlaneMachineSet` will
+take over the management.
+We enforce this recommendation so that users can be confident in the Machine `providerSpec` that they have configured
+and that Machine API will be able to create valid Machines from the spec before they then start using our automation.
+
+##### Why not to populate the spec for the customer
+
+One idea posed, is to allow the `ControlPlaneMachineSet` to be automatically populated if the spec is left blank when it
+is created. This would improve the UX by allowing customers not to have to worry about the spec and allow it to be
+inferred from the existing Machines within the cluster. It would also make it easier for managed services to adopt
+`ControlPlaneMachineSets` throughout the fleet without having to manually check each cluster.
+
+There are however a few concerns about this idea which means we are planning to not implement this (at least for the
+first iteration of the project):
+- It may encourage users to attempt to use `ControlPlaneMachineSets` with UPI clusters which are incorrectly configured
+  - Some users have Machines in their cluster, that aren't actually configured correctly, therefore we would be creating
+    an incorrectly configured `ControlPlaneMachineSet` for them, it is then unclear who is at fault for this and we
+    could end up with an increased number of support tickets due to these misconfigurations
+  - As an example, this is particularly prevalent in UPI clusters where users can forget to remove the Machine manifests
+    from the install manifest directory, this has been the source of many bugs where users have had Machines stuck in
+    Provisioning for an extended period. Inferring the specs from these Machines would certainly make the
+    `ControlPlaneMachineSet` invalid.
+- Some users may have made out of band changes to their Control Plane Machines which are not reflected within the
+  Machine specs
+  - We document the procedure within product docs of replacing a Control Plane Machine, though it is a manual process
```

Comment:
I am guessing that at least for now, we will want to keep the manual steps too, and perhaps just highlight that this feature will make the process much easier for supported configurations?

Reply:
I think we will want to have a manual process and a, with a ControlPlaneMachineSet. And if we can cross reference so that users know about the new type in the manual process, then great, that makes things a lot easier :)